src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -138,24 +140,28 @@ public int getValue(){
     private static final String R_SCRIPT = "insertSizeHistogram.R";
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public boolean requiresReads(){return true;}
+
+    @Override
+    public void runTool(final JavaSparkContext ctx) {
 
         final JavaRDD<GATKRead> filteredReads = getReads();
 
+        final SAMFileHeader readsHeader = getHeaderForReads();
+        final String inputFileName = getReadSourceName();
+
         // Class where real metric-collection work is delegated to.
         final InsertSizeMetricsCollectorSpark collector = new InsertSizeMetricsCollectorSpark(filteredReads,
-                                                                                              // METRIC_ACCUMULATION_LEVEL,
-                                                                                              getHeaderForReads().getReadGroups(),
+                                                                                              readsHeader,
+                                                                                              METRIC_ACCUMULATION_LEVEL,
                                                                                               DEVIATIONS_TOL);
 
-        final SAMFileHeader readsHeader = getHeaderForReads();
-        final String inputFileName = getReadSourceName();
-
+        // similar to Picard::CollectInsertSizeMetrics.finish()
         if(OUTPUT != null) {
-            writeMetricsFile(collector, readsHeader, inputFileName);
```

Comment:
This particular class is the command line tool that drives the collection process. When it's run from the command line it'll run the actual algorithm (that's in InsertSizeMetricsCollectorSpark), but then not do anything with the results (in memory) and then the command will exit and the metrics will be lost. If in the future we want other tools to invoke the same process (rather than reading the results from the metrics file), they can use InsertSizeMetricsCollectorSpark itself, but they wouldn't want to invoke the command line tool through this command line driver class itself. They'd be Spark tools in their own right and would invoke the InsertSizeMetricsCollector with their own collection of reads and spark context, etc. As it stands in this PR, invoking this tool without an output will do a lot of work but not return any results, right?